### PR TITLE
feat: add total field to UserTopReader to get the users total amont of top reader badges

### PR DESCRIPTION
**Pull request by omBratteng**
AS-661

### CODE REVIEW

File: src/graphorm/index.ts [hunk 1]
```diff
@@ -203,6 +204,14 @@ const obj = new GraphORM({
...
         },
       },
+      total: {
+        select: (_, alias, qb) =>
+          qb
+            .select('COALESCE(COUNT(*), 0)')
+            .from(UserTopReader, 'utp')
+            .where(`utp."userId" = ${alias}."userId"`)
+            .groupBy('utp."userId"'),
+      },
```
Comment (omBratteng) [inline src/graphorm/index.ts:214]: I went with the easiest solution here, and just do count directly of the table.
Alternatively my other plan was to extend the UserStats MV to include the count of total number of badges.

Let me know if I should go the MV route

Comment (omBratteng) [reply src/graphorm/index.ts:214]: Going to use MV for this

**Review by rebelchris — APPROVED** [review]
Happy with the MV route, but for now this also works for me.

File: src/graphorm/index.ts (hunk 1 above)
Comment (omBratteng) [reply src/graphorm/index.ts]: Refactored to MV

**Review by rebelchris — APPROVED** [review]
Works for me, but wait with merging for Ante's approval please :)

File: src/schema/users.ts [hunk 2]
```diff
@@ -1699,7 +1704,7 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
     },
     topReaderBadge: async (
       _,
-      { limit = 5 }: { limit: number },
+      { limit = 5, userId }: { limit: number; userId?: string },
```
Comment (capJavert) [inline src/schema/users.ts:1707]:
```suggestion
      { limit = 5, userId }: { limit: number; userId: string },
```

File: src/schema/users.ts [hunk 3]
```diff
@@ -1709,7 +1714,7 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         (builder) => {
           builder.queryBuilder = builder.queryBuilder
             .andWhere(`${builder.alias}.userId = :userId`, {
-              userId: ctx.userId,
+              userId: userId ?? ctx.userId,
```
[the remark this answers is not in the document]
Comment (omBratteng) [reply src/schema/users.ts:1723]: Yeah, can do that

**Review by sshanzel — COMMENTED** [review]
Don't have much to add. Looks great! 🚀

File: src/schema/users.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (omBratteng) [reply src/schema/users.ts:1723]:
```
```suggestion
              userId: userId,
```

File: src/schema/users.ts [hunk 4]
```diff
@@ -1715,7 +1720,7 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         (builder) => {
           builder.queryBuilder = builder.queryBuilder
             .andWhere(`${builder.alias}.userId = :userId`, {
-              userId: ctx.userId,
+              userId: userId,
```
Comment (capJavert) [inline src/schema/users.ts:1723]:
```suggestion
              userId,
```

**Review by capJavert — APPROVED** [review]
As we spoke, just watch for MV recreation, might be expensive so keep an eye out.

**Review by rebelchris — APPROVED** [review]

**Pull request merged**